db/schema/patch-008-009.sql

```diff
@@ -0,0 +1,278 @@
+
+-- A new table to hold a history of account lifecycle events.
+-- Events are added in simple increasing auto-increment order.
+CREATE TABLE eventLog (
+  seqnum BIGINT UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  uid BINARY(16) NOT NULL,
+  typ ENUM('create', 'verify', 'reset', 'delete') NOT NULL,
+  iat INT UNSIGNED NOT NULL
+) ENGINE=InnoDB;
+
+
+-- When an account is created, log a "create" event.
+-- Pre-verified accounts may also get a "verify" event.
+CREATE PROCEDURE `createAccount_2` (
+    IN `uid` BINARY(16) ,
+    IN `normalizedEmail` VARCHAR(255),
+    IN `email` VARCHAR(255),
+    IN `emailCode` BINARY(16),
+    IN `emailVerified` TINYINT(1),
+    IN `kA` BINARY(32),
+    IN `wrapWrapKb` BINARY(32),
+    IN `authSalt` BINARY(32),
+    IN `verifierVersion` TINYINT UNSIGNED,
+    IN `verifyHash` BINARY(32),
+    IN `verifierSetAt` BIGINT UNSIGNED,
+    IN `createdAt` BIGINT UNSIGNED,
+    IN `locale` VARCHAR(255)
+)
+BEGIN
+    DECLARE EXIT HANDLER FOR SQLEXCEPTION
+    BEGIN
+        ROLLBACK;
+        RESIGNAL;
+    END;
+
+    START TRANSACTION;
+
+    INSERT INTO accounts(
+        uid,
+        normalizedEmail,
+        email,
+        emailCode,
+        emailVerified,
+        kA,
+        wrapWrapKb,
+        authSalt,
+        verifierVersion,
+        verifyHash,
+        verifierSetAt,
+        createdAt,
+        locale
+    )
+    VALUES(
+        uid,
+        LOWER(normalizedEmail),
+        email,
+        emailCode,
+        emailVerified,
+        kA,
+        wrapWrapKb,
+        authSalt,
+        verifierVersion,
+        verifyHash,
+        verifierSetAt,
+        createdAt,
+        locale
+    );
+
+    INSERT INTO eventLog(
+        seqnum,
+        uid,
+        typ,
+        iat
+    )
+    VALUES(
+        NULL,
```

Comment:
Is this to be explicit for some reason? Otherwise this will do.

```sql
INSERT INTO eventLog(uid, typ, iat) VALUES (uid, "create", UNIX_TIMESTAMP());
```